src/Microsoft.Health.Dicom.Api/Features/Routing/UrlResolver.cs

```diff
@@ -146,11 +151,45 @@ private Uri RouteUri(string routeName, RouteValueDictionary routeValues)
     {
         HttpRequest request = _httpContextAccessor.HttpContext.Request;
 
-        return new Uri(
-            UrlHelper.RouteUrl(
+        return GetRouteUri(
+                ActionContext.HttpContext,
                 routeName,
                 routeValues,
                 request.Scheme,
-                request.Host.Value));
+                request.Host.Value);
+    }
+
+    private Uri GetRouteUri(HttpContext httpContext, string routeName, RouteValueDictionary routeValues, string scheme, string host)
+    {
+        var uriString = string.Empty;
+
+        if (httpContext == null)
+        {
+            uriString = UrlHelper.RouteUrl(
+                routeName,
+                routeValues,
+                scheme,
+                host);
+        }
+        else
+        {
+            // NOTE: Append "/" to the end of the path base to workaround a known bug in UrlHelper.RouteUrl for endpoint routing.
+            //       Remove the workaround when we pick up the fix. (https://github.com/dotnet/aspnetcore/issues/53177)
+            var pathBase = httpContext.Request?.PathBase.ToString();
+            if (!string.IsNullOrEmpty(pathBase) && (string.IsNullOrEmpty(httpContext.Request?.Path) || string.Equals(httpContext.Request?.Path, "/", StringComparison.Ordinal)))
```

Comment:
In what case is `Path` empty? According to this issue, it doesn't actually appear to be a bug